Tests/MLXTests/MLXArrayTests.swift

```diff
@@ -102,10 +102,79 @@ class MLXArrayTests: XCTestCase {
         let expected: [Int32] = [0, 2, 4, 6, 8, 10, 12, 14]
         assertEqual(s, MLXArray(expected, [4, 2]))
 
-        XCTAssertEqual(s.strides, [4, 2])
+        XCTAssertEqual(s.internalStrides, [4, 2])
 
         let s_arr = s.asArray(Int32.self)
         XCTAssertEqual(s_arr, expected)
     }
 
+    func testContiguousStrides() {
+        XCTAssertEqual(contiguousStrides(shape: [1, 1, 1]), [1, 1, 1])
+        XCTAssertEqual(contiguousStrides(shape: [4, 4]), [4, 1])
+        XCTAssertEqual(contiguousStrides(shape: [4, 2]), [2, 1])
+        XCTAssertEqual(contiguousStrides(shape: [3, 2, 1, 5]), [10, 5, 5, 1])
+    }
+
+    func testAsDataContiguous() {
+        // contiguous source
+        let a = MLXArray(0 ..< 16, [4, 4])
+
+        let result = a.asData()
+        XCTAssertEqual(result.shape, [4, 4])
+        XCTAssertEqual(result.strides, [4, 1])
+        XCTAssertEqual(result.dType, .int32)
+    }
+
+    func testAsDataContiguousNoCopy() {
+        // contiguous source
+        let a = MLXArray(0 ..< 16, [4, 4])
+
+        do {
+            let result = a.asData(access: .noCopy)
+            XCTAssertEqual(result.shape, [4, 4])
+            XCTAssertEqual(result.strides, [4, 1])
+            XCTAssertEqual(result.dType, .int32)
+        }
+        do {
+            let result = a.asData(access: .noCopyIfContiguous)
+            XCTAssertEqual(result.shape, [4, 4])
+            XCTAssertEqual(result.strides, [4, 1])
+            XCTAssertEqual(result.dType, .int32)
+        }
+    }
+
+    func testAsDataNonContiguous() {
```

Comment:
The remaining tests are the interesting ones -- they deal with non contiguous backing and make sure they behave as expected.